src/main/app/scripts/cache-busting.js

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env node
+
+const fs = require("fs");
+const path = require("path");
+const crypto = require("crypto");
+
+/**
+ * Generates a hash for translation files
+ */
+function generateTranslationHash() {
+  const i18nDir = path.join(__dirname, "../src/assets/i18n");
+  const files = ["en.json", "nl.json"];
+
+  let combinedContent = "";
+
+  files.forEach((file) => {
+    const filePath = path.join(i18nDir, file);
+    if (fs.existsSync(filePath)) {
+      const content = fs.readFileSync(filePath, "utf8");
+      combinedContent += content;
+    }
+  });
+
+  // Generate a short hash from the combined content
+  return crypto
+    .createHash("md5")
+    .update(combinedContent)
+    .digest("hex")
+    .substring(0, 8);
+}
+
+/**
+ * Generates a hash for font files
+ */
+function generateFontHash() {
+  const assetsDir = path.join(__dirname, "../src/assets");
+  const fontFiles = [
+    "MaterialSymbolsOutlined.woff2",
+    "fonts/Roboto/300.woff2",
+    "fonts/Roboto/400.woff2",
+    "fonts/Roboto/500.woff2",
+  ];
+
+  let combinedContent = "";
+
+  fontFiles.forEach((file) => {
+    const filePath = path.join(assetsDir, file);
+    if (fs.existsSync(filePath)) {
+      const stats = fs.statSync(filePath);
+      // Use file size and modification time for fonts (binary files)
+      combinedContent += `${file}:${stats.size}:${stats.mtime.getTime()}`;
+    }
+  });
+
+  return crypto
+    .createHash("md5")
+    .update(combinedContent)
+    .digest("hex")
+    .substring(0, 8);
+}
+
+/**
+ * Replaces BUILD_HASH placeholder with actual hash in the compiled JavaScript
+ */
+function replaceBuildHash() {
+  const distDir = path.join(__dirname, "../dist/zaakafhandelcomponent");
+
+  if (!fs.existsSync(distDir)) {
+    console.log("Dist directory not found, skipping hash replacement");
+    return;
+  }
+
+  const translationHash = generateTranslationHash();
+  const fontHash = generateFontHash();
+
+  console.log(`Translation hash: ${translationHash}`);
+  console.log(`Font hash: ${fontHash}`);
+
+  // Find all JavaScript files in the dist directory
+  function processDirectory(dir) {
+    const files = fs.readdirSync(dir);
+
+    files.forEach((file) => {
+      const filePath = path.join(dir, file);
+      const stat = fs.statSync(filePath);
+
+      if (stat.isDirectory()) {
+        processDirectory(filePath);
+      } else if (file.endsWith(".js") || file.endsWith(".html")) {
+        let content = fs.readFileSync(filePath, "utf8");
+        let fileUpdated = false;
+
+        // Replace BUILD_HASH with actual hash (only in JS files)
+        if (file.endsWith(".js") && content.includes("BUILD_HASH")) {
```

Comment:
We already hash all JS files using the standard Angular functionality for this no? `"outputHashing": "all",` in angular.json. Do you want to replace that with this solution? If so, `outputHashing` should probably be turned off in `angular.json` or not?